### PR TITLE
ref(js): Remove UNSAFE_componentWillReceiveProps from deprecatedforms/PasswordField

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/components/deprecatedforms/passwordField.tsx [hunk 1]
```diff
@@ -27,17 +27,15 @@ export default class PasswordField extends InputField<Props, State> {
...
-  UNSAFE_componentWillReceiveProps(nextProps: Props) {
+  componentDidUpdate(prevProps: Props) {
     // close edit mode after successful save
     // TODO(dcramer): this needs to work with this.context.form
     if (
       this.props.formState &&
-      this.props.formState === FormState.SAVING &&
-      nextProps.formState === FormState.READY
+      prevProps.formState === FormState.SAVING &&
+      this.props.formState === FormState.READY
```
Comment (k-fish) [inline static/app/components/deprecatedforms/passwordField.tsx:36]: Maybe add a check that `editing` isn't already `false`?

Comment (evanpurkhiser) [reply static/app/components/deprecatedforms/passwordField.tsx:36]: it will just be a no-op then, I think it doesn't matter

Comment (evanpurkhiser) [reply static/app/components/deprecatedforms/passwordField.tsx:36]: I think this one is good to go looking over it again

Comment (k-fish) [reply static/app/components/deprecatedforms/passwordField.tsx:36]: iirc it's only primitives w/ useState hook that are a no-op, but if you've check it and it seems to work then 👍

**Review by k-fish — APPROVED** [review]
Given that you can confirm it's a no-op then 👍

**Pull request merged**